### PR TITLE
fixed assertion methods

### PR DESCRIPTION
**Pull request by trkohler**
This pull request addresses issue #2835 

- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``flake8`` locally, and
understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [ ] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

### CODE REVIEW

File: Tests/test_SearchIO_model.py [hunk 1]
```diff
@@ -319,7 +319,7 @@ def test_getitem_alt_ok(self):
         query = QueryResult([hit11])
         self.assertEqual(hit11, query["hit1"])
         self.assertEqual(hit11, query["alt1"])
-        self.assertTrue(hit11.id != "alt1")
+        self.assertEqual(hit11.id, "alt1")
```
Comment (bow) [inline Tests/test_SearchIO_model.py:322]: The comparison here becomes incorrect. It should be `assertNotEqual`

File: Tests/test_SearchIO_model.py [hunk 2]
```diff
@@ -196,8 +196,8 @@ def test_setitem_ok_alt(self):
         self.assertEqual(hit11, query["hit1"])
         self.assertEqual(hit11, query["alt1"])
         self.assertEqual(hit11, query["alt11"])
-        self.assertTrue(hit11.id != "alt1")
-        self.assertTrue(hit11.id != "alt11")
+        self.assertEqual(hit11.id, "alt1")
+        self.assertEqual(hit11.id, "alt11")
```
Comment (bow) [inline Tests/test_SearchIO_model.py:200]: These should also be `assertNotEqual`

File: Tests/test_Restriction.py [hunk 3]
```diff
@@ -230,7 +230,7 @@ def test_comparisons(self):
         # Comparison of iso- and neoschizomers
         self.assertEqual(Acc65I, Acc65I)
         self.assertNotEqual(Acc65I, KpnI)
-        self.assertFalse(Acc65I == Asp718I)
+        self.assertNotEqual(Acc65I, Asp718I)
         self.assertFalse(Acc65I != Asp718I)
```
Comment (bow) [inline Tests/test_Restriction.py:234]: There `assertFalse` call looks like it can also be updated in this PR. Would it make sense to update it?

**Comment by trkohler** [comment]
may I break something? Seems like code didn't pass CI/CD flow

File: Tests/test_Restriction.py (hunk 3 above)
Comment (trkohler) [reply Tests/test_Restriction.py:234]: 233 line brought me a mistake.
Seems like **Acc65I == Asp718I**, so **self.assertNotEqual(Acc65I, Asp718I)** is not correct.

234 line contains the opposite statement. **self.assertEqual(Acc65I, Asp718I)**

May I remove 233?

Screen Shot 2020-05-01 at 10 11 44 AM

Comment (trkohler) [reply Tests/test_Restriction.py:234]: As I understand, **assertFalse(something not equal)**  in 234 line should become to **assertEqual**

But, when I change:
`self.assertNotEqual(Acc65I, Asp718I)` <- 233 line
`self.assertEqual(Acc65I, Asp718I)` <- 234 line

I got an error in the 233 line.

If I comment it out, assertion error happens on the next line 🤔

Screen Shot 2020-05-01 at 10 11 44 AM

Comment (trkohler) [reply Tests/test_Restriction.py:234]: I pass local tests only if I remain **test_comparisons** in **test_Restriction.py** the same.

Comment (peterjc) [reply Tests/test_Restriction.py:235]: I thought test_Restriction.py was done, see discussion on #2833 - this is a strange case where the objects define conflicting comparison methods.

**Comment by peterjc** [comment]
Thank you - that's most of this issue done now :)

**Pull request merged**